src/node_http2.cc

```diff
@@ -55,10 +55,27 @@ enum Http2OptionsIndex {
   IDX_OPTIONS_FLAGS
 };
 
+static uint32_t http2_padding_buffer[3];
+static uint32_t http2_options_buffer[IDX_OPTIONS_FLAGS + 1];
+static uint32_t http2_settings_buffer[IDX_SETTINGS_COUNT + 1];
+static double http2_session_state_buffer[IDX_SESSION_STATE_COUNT];
+static double http2_stream_state_buffer[IDX_STREAM_STATE_COUNT];
+
+static const size_t http2_options_buffer_byte_length =
+    sizeof(http2_options_buffer) * (IDX_OPTIONS_FLAGS + 1);
+static const size_t http2_settings_buffer_byte_length =
+    sizeof(http2_settings_buffer) * (IDX_SETTINGS_COUNT + 1);
+static const size_t http2_padding_buffer_byte_length =
+    sizeof(http2_padding_buffer) * 3;
+static const size_t http2_stream_state_buffer_byte_length =
+    sizeof(http2_stream_state_buffer) * IDX_STREAM_STATE_COUNT;
+static const size_t http2_session_state_buffer_byte_length =
+    sizeof(http2_session_state_buffer) * IDX_SESSION_STATE_COUNT;
```

Comment:
By the way, what I had in mind is not removing them from the environment but instead making these static-length array members of `Environment`. I am not sure whether this kind of usage is going to be a problem for embedders.